Prefer "number of online processors" since some cpus will overreport
the total number of cpus.

See: https://pari.math.u-bordeaux.fr/cgi-bin/bugreport.cgi?bug=2593#25

--- a/src/mt/pthread.c
+++ b/src/mt/pthread.c
@@ -139,7 +139,9 @@ void mt_broadcast(GEN code) {(void) code;}
 void pari_mt_init(void)
 {
   pari_mt = NULL;
-#ifdef _SC_NPROCESSORS_CONF
+#ifdef _SC_NPROCESSORS_ONLN
+  if (!pari_mt_nbthreads) pari_mt_nbthreads = sysconf(_SC_NPROCESSORS_ONLN);
+#elif _SC_NPROCESSORS_CONF
   if (!pari_mt_nbthreads) pari_mt_nbthreads = sysconf(_SC_NPROCESSORS_CONF);
 #elif defined(_WIN32)
   if (!pari_mt_nbthreads) pari_mt_nbthreads = win32_nbthreads();
